### PR TITLE
🐛 Previously published drafts ordering

### LINKED ISSUE #8495: Previously published drafts are listed before new drafts
### Issue Summary

Any drafts that were previously published or have had their "published at" date manually set will be ordered above new drafts. In this example you can see the newly created draft appearing 2nd in the list despite the top post being created a few days ago and having been last edited a few hours earlier than the new post was created:

screen shot 2017-05-30 at 15 59 57

### Steps to Reproduce

1. Create a new post
2. Publish the post then unpublish it to make it a draft again
3. Go the the posts list, note that it's at the top of the list
4. Create a new post and keep it as a draft
5. Go back to the posts list, note that your new post is in 2nd position rather than 1st position

### Technical details:

* Ghost Version: 1.0 alpha 21
* Node Version: 6.10.1
* Database: SQLIte

### PR DESCRIPTION
**Pull request by naz**
closes #8495

- Fixes ordering of posts when one of the drafts was previously published 
- Adds test coverage for the case above
- Refactored posts integration tests

### CODE REVIEW

**Comment by gargol** [comment]
@kirrg001 updated the tests, it's ready for another review :wink:

File: core/test/integration/api/api_posts_spec.js [hunk 1]
```diff
@@ -546,6 +547,50 @@ describe('Post API', function () {
                 done();
             }).catch(done);
         });
+
+        it('can fetch all posts with correct order when unpublished draft is present', function (done) {
```
Comment (kirrg001) [inline core/test/integration/api/api_posts_spec.js]: We try to avoid using the done callback if not really necessary.

**Comment by kirrg001** [comment]
Suggestion from https://github.com/TryGhost/Ghost/issues/8495#issuecomment-304926384 works 👍 👻

File: core/test/integration/api/api_posts_spec.js (hunk 1 above)
Comment (kirrg001) [reply core/test/integration/api/api_posts_spec.js]: Up to you (depends if you want to spend time on it), but in separate commit 👍

**Comment by gargol** [comment]
@kirrg001 updated commits. Let me know if there's anything else to be done before merging these in :)

File: core/test/integration/api/api_posts_spec.js [hunk 2]
```diff
@@ -18,9 +16,9 @@ describe('Post API', function () {
     var localSettingsCache = {};
 
     before(testUtils.teardown);
-    after(testUtils.teardown);
+    afterEach(testUtils.teardown);
 
-    before(testUtils.setup('users:roles', 'perms:post', 'perms:init', 'posts'));
+    beforeEach(testUtils.setup('users:roles', 'perms:post', 'perms:init', 'posts'));
```
Comment (kirrg001) [inline core/test/integration/api/api_posts_spec.js]: We now reset the whole db after each test. Isn't it enough to reset the db after the test case you have added? 🤔

Comment (gargol) [reply core/test/integration/api/api_posts_spec.js]: Good :fishing_pole_and_fish: Tested without `Each` on current suite and seems to work. Reverted this change

Comment (kirrg001) [reply core/test/integration/api/api_posts_spec.js]: 👍

**Pull request merged**